mesonbuild/minstall.py

```diff
@@ -584,6 +609,16 @@ def install_data(self, d: InstallData, dm: DirMaker, destdir: str, fullprefix: s
                 self.did_install_something = True
             self.set_mode(outfilename, i.install_mode, d.install_umask)
 
+    def install_symlinks(self, d: InstallData, dm: DirMaker, destdir: str, fullprefix: str) -> None:
+        for s in d.symlinks:
+            if not self.should_install(s):
+                continue
+            full_dst_dir = get_destdir_path(destdir, fullprefix, s.install_path)
+            full_link_name = get_destdir_path(destdir, fullprefix, s.name)
+            dm.makedirs(full_dst_dir, exist_ok=True)
+            if self.do_symlink(s.target, full_link_name, full_dst_dir):
```

Comment:
Part of the functional effects here are done in a do_symlink function, part in install_symlink. I wonder whether we actually need two functions at all? The only other secondary functions we have are do_copyfile and do_copydir, but in both cases we need to use them multiple times, so they are factored out for practical reasons.

Reply:
We do in fact have some code for installing targets that handles symlinks too. Specifically shared library soname aliases. So I think it makes some sense to factor out that code into a do_symlink method, and take advantage of a single-source handling of remove + symlink + append_to_log + warn when symlink creation doesn't work on a given plaform.

Reply:
I added a last commit so that install_targets also uses the `do_symlink` function. Initially I just created it for completion. Good to know it was actually useful and not just stupid :D